lib/os/mempool.c

```diff
@@ -328,8 +329,13 @@ void *sys_mem_pool_alloc(struct sys_mem_pool *p, size_t size)
 	struct sys_mem_pool_block *blk;
 	uint32_t level, block;
 	char *ret;
+	int lock_ret;
 
-	sys_mutex_lock(&p->mutex, K_FOREVER);
+	lock_ret = sys_mutex_lock(&p->mutex, K_FOREVER);
+	CHECKIF(lock_ret != 0) {
+		ret = NULL;
+		goto out;
```

Comment:
you should return here and not jump to `out` to avoid call `sys_mutex_unlock` since the mutex was not locked.

Reply:
Changed goto to return NULL and changed the other goto to just return, could you take another look?